deployment/api_server.py

```
@@ -102,4 +102,7 @@ def get_emotions():
     print("  - GET  /emotions - List emotions")
     print("=" * 50)
     
-    app.run(host='0.0.0.0', port=5000, debug=False)
+    # Use environment variable for host binding, default to localhost for security
+    host = os.getenv('API_HOST', '127.0.0.1')
```

Ruff (0.12.2)

Undefined name `os`

(F821)